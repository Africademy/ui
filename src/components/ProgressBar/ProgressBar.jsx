import React from 'react';
import PropTypes from 'prop-types';

import {
  ContainerStyled,
  ProgressStyled,
} from './style';

const ProgressBar = ({
  progress,
}) => (
  <ContainerStyled>
    <ProgressStyled progress={progress} />
  </ContainerStyled>
);

ProgressBar.propTypes = {
  /** Percentage of progress. */
  progress: PropTypes.string.isRequired,
};

export default ProgressBar;
