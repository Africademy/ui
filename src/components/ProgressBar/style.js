import styled from 'styled-components';
import { gray, grayLight } from '../style/colors';

export const ContainerStyled = styled.div`
  width: ${props => (props.fullWidth ? '100%' : '140px')};
  height: 8px;
  background-color: ${grayLight};
  border-radius: 4px;
  margin: 16px 0px;
`;

export const ProgressStyled = styled.div`
  background-color: ${gray};
  height: 100%;
  width: ${props => props.progress || '100%'};
  border-radius: 4px;
`;
