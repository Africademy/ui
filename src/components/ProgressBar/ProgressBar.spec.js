import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import ProgressBar from './ProgressBar';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('ProgressBar', ()=> {
    test('gestWidth should retturn 250', ()=> {
        const wrapper = shallow(<ProgressBar percent={50} width={500}/>)
        const width = wrapper.instance().getWidthAsPercentOfTotalWidth()

        expect(width).toEqual(250);
    })
})