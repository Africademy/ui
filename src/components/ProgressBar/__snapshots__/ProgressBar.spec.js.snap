// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > ProgressBar Matches snapshot when passed only required props 1`] = `
<styled.div>
  <styled.div
    progress="jest-auto-snapshots String Fixture"
  />
</styled.div>
`;
